### DISCUSSION TITLE
read code have some confusion

### DISCUSSION THREAD

**Discussion by zhongmax** [pypa/setuptools#2791] [Q&A] [answered]
Recently I read pip 20.3.3 resource code, I want to know how pip parse tar.gz or whl depends
when I read `pip/_vendor/pkg_resources/__init__.py` line 2561
```python
class Distribution:
    """Wrap an actual or potential sys.path entry w/metadata"""
    PKG_INFO = 'PKG-INFO'

    def __init__(
            self, location=None, metadata=None, project_name=None,
            version=None, py_version=PY_MAJOR, platform=None,
            precedence=EGG_DIST):
        self.project_name = safe_name(project_name or 'Unknown')
        if version is not None:
            self._version = safe_version(version)
        self.py_version = py_version
        self.platform = platform
        self.location = location
        self.precedence = precedence
        self._provider = metadata or empty_provider
        print "Distribution init"
```
![image](https://user-images.githubusercontent.com/20065316/133070903-edf02b94-fafc-407f-80f3-dfb523a612d3.png)

when line 2576 is executed, I found self object Protected Attributes `_dep_map` have parsed value, I want to konw the flow of this execution. Thank you!
![image](https://user-images.githubusercontent.com/20065316/133073232-e59472d1-533d-4986-a704-8af63edde83e.png)

**Comment by webknjaz** [comment, marked as answer]
It's here
https://github.com/pypa/setuptools/blob/4bce6d5/pkg_resources/__init__.py#L2698-L2701
